batcher-bot/liquidity/src/provision.ts

```diff
@@ -0,0 +1,240 @@
+import {
+  batch_provision,
+  order,
+  token_pair,
+  contract_details,
+  token,
+} from "./types";
+import { Option } from "prelude-ts";
+
+const provision = new Map<number, batch_provision>();
+
+const parse_tolerance = (side: number, tol: string): number => {
+  if (side == 0) {
+    if (tol == "worse") {
+      return 0;
+    }
+
+    if (tol == "better") {
+      return 2;
+    }
+
+    return 1;
+  } else {
+    if (tol == "worse") {
+      return 2;
+    }
+
+    if (tol == "better") {
+      return 0;
+    }
+
+    return 1;
+  }
+};
+
+const provision_sell = (
+  batch_number: number,
+  token_pair: token_pair,
+  order: order
+): Option<order> => {
+  let existing_buy_provision = 0;
+  let existing_sell_provision = 0;
+
+  let existing_provision = provision.get(batch_number);
+  try {
+    if (existing_provision) {
+      existing_buy_provision = existing_provision.buy_side_provision;
+      existing_sell_provision = existing_provision.sell_side_provision;
+    }
+  } catch {}
+
+  if (token_pair.side == "either" && existing_buy_provision > 0) {
+    return Option.none();
+  }
+
+  let upper_provision_bound = token_pair.sell_limit_per_batch;
+  let remaining = upper_provision_bound - existing_sell_provision;
+
+  if (remaining <= 0) {
+    return Option.none();
+  }
+
+  let sell_decimals = order.swap.to.decimals;
+  let scaled_amount = remaining * 10 ** sell_decimals;
+
+  let prov_order: order = {
+    swap: {
+      from: {
+        token: order.swap.to,
+        amount: scaled_amount,
+      },
+      to: order.swap.from.token,
+    },
+    side: 1,
+    tolerance: parse_tolerance(1, token_pair.sell_tolerance),
+  };
+  let updated_provision: batch_provision = {
+    batch_number: batch_number,
+    buy_side_provision: existing_buy_provision,
+    sell_side_provision: existing_sell_provision + remaining,
+  };
```

Comment:
Small optimizations, but all those `let` can be turned into `const`. Basically V8 has a better memory management with `const`, and thus, a smaller footprint

Reply:
Thanks, fixed